model/patch_lifecycle.go

```diff
@@ -685,13 +680,6 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 		ppStorageMethod = evergreen.ProjectStorageMethodDB
 	}
 
-	if p.PatchedParserProject != "" {
```

Comment:
GH doesn't let me comment on unchanged lines, but the comment on L677 has to be removed since the parser project is never stored as a string in the patch doc anymore.

Reply:
The fallback of setting a default storage method on L685 also should be removed because [the storage method is always set on patch creation for newer patches](https://github.com/evergreen-ci/evergreen/blob/b029ef66fa020c5c19294224103c4af794adc510/units/patch_intent.go#L376-L380), and we're assuming that nobody's going to finalize old patches from before March 2023. The version document can just inherit the patch document's project storage method as-is.

Reply:
done